packages/components/src/styles/components/dialog-primitive.scss

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// DIALOG PRIMITIVE
+//
+
+$hds-dialog-primitive-horizontal-padding: 24px;
+$hds-dialog-primitive-vertical-padding: 16px;
+
+
+// WRAPPER CONTAINER (<dialog>)
+
+// The special declaration below is used to allow the `<dialog>` to unset
+// all its predefined styles and set them at `Modal/Flyout` level
+//
+// Since the `:where()` selector has a specificity of `0` (see https://developer.mozilla.org/en-US/docs/Web/CSS/:where)
+// the styles declared below are applied to the `<dialog>` element, but are overwritten by any other style applied via classname
+//
+
+:where(.hds-dialog-primitive__wrapper) {
+  // -----------
+  all: unset; // removes default dialog & inherited styles. Must be the first style to be applied
+  // -----------
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  height: 100%;
+  margin: 0;
+  padding: 0;
+  background: var(--token-color-surface-primary);
+}
+
+
+// Note: this wraps both .hds-dialog-primitive__header & .hds-dialog-primitive__description
+.hds-dialog-primitive__wrapper-header {
+  flex: none;
+  background: var(--token-color-surface-faint);
+  border-bottom: 1px solid var(--token-color-border-primary);
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+}
+
+.hds-dialog-primitive__wrapper-body {
+  flex: 1 1 auto;
+  // this is necessary, otherwise if there are words longer than the width of the available space for the list "item" content
+  overflow-x: hidden;
+  // we want the content to vertically scroll if needed
+  overflow-y: auto;
+  overscroll-behavior: contain;
+}
+
+.hds-dialog-primitive__wrapper-footer {
+  flex: none;
+  border-top: 1px solid var(--token-color-border-primary);
+  border-bottom-right-radius: inherit;
+  border-bottom-left-radius: inherit;
+}
+
+
+// HEADER
+
+.hds-dialog-primitive__header {
+  display: flex;
+  gap: 16px;
+  align-items: flex-start;
+  padding: $hds-dialog-primitive-vertical-padding $hds-dialog-primitive-horizontal-padding;
+  color: var(--token-color-foreground-strong); // default color (applied to both the title and the icon)
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+}
+
+.hds-dialog-primitive__icon {
+  flex: none;
+  align-self: center;
+}
+
+.hds-dialog-primitive__title {
+  flex-grow: 1;
+}
+
+.hds-dialog-primitive__tagline {
+  margin-bottom: 4px;
+  color: var(--token-color-foreground-faint); // default color
+}
+
+.hds-dialog-primitive__dismiss {
+  align-self: center;
```

Comment:
We should top align the dismiss button (instead of center). Upon further investigation, it looks like the current figma implementation of modal and flyout had different behaviors (center for modal, top aligned for flyout). I'm aligning them to the top in this branch: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/branch/Yo8J7OzRaIBA4ytBPPDEqh/HDS-Product---Components?type=design&node-id=53258%3A135&mode=design&t=gGHcTGqSRTFmxDVT-1

Reply:
I pushed an update

Reply:
@KristinLBradley I don't see the commit 🤔

Reply:
@majedelass @KristinLBradley this will cause a visual mis-alignment in the modal, when there's no tagline (which is the most common use case)

<img width="616" alt="screenshot_3501" src="https://github.com/hashicorp/design-system/assets/686239/0e79a8ba-62a5-45f4-af65-3a543034cad3">

Reply:
Yeah that's fine, that currently exists in the figma design too: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/branch/Yo8J7OzRaIBA4ytBPPDEqh/HDS-Product---Components?type=design&node-id=22928%3A56204&mode=design&t=RIpe2DJSgIZS5TKY-1

Reply:
Also current day flyout behaves this way: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?type=design&node-id=24759%3A104119&mode=design&t=WbToqy24HvmSKjqw-1

Reply:
> current day flyout behaves this way

Not in code, from what I see:
<img width="1332" alt="screenshot_3503" src="https://github.com/hashicorp/design-system/assets/686239/88c4bb86-c562-464f-bdbe-f34e32174a26">

@jorytindall do you remember if it was an intentional decision?

Reply:
Yeah both modal and flyout behaved differently _in figma_ and wanted to align them moving forward. I did discuss with the design team that it's beneficial for the dismiss button to align to the top for its placement to be predictable since it's an actional UI piece.

Reply:
we had quite a bit of feedback from consumers on what they considered being 'misaligned' items so I would be surprised if they won't call this out

Reply:
+1 on @alex-ju's comment

Reply:
I pushed it to the wrong branch :( 
https://github.com/hashicorp/design-system/pull/1945/commits

Reply:
@KristinLBradley [all designers agree on aligning the icon to the top](https://hashicorp.slack.com/archives/C025N5V4PFZ/p1707847200682889). Can you port the change in this branch, so I can approve?

Reply:
I moved the commit to this branch now.